pkg/consensusfault/check.go

```diff
@@ -138,15 +144,36 @@ func (s *ConsensusFaultChecker) VerifyConsensusFault(ctx context.Context, h1, h2
 }
 
 // Checks whether a block header is correctly signed in the context of the parent state to which it refers.
-func verifyBlockSignature(ctx context.Context, view FaultStateView, blk types.BlockHeader, nv network.Version) error {
-	minerInfo, err := view.MinerInfo(ctx, blk.Miner, nv)
+func verifyBlockSignature(ctx context.Context, blk types.BlockHeader, nv network.Version, curEpoch abi.ChainEpoch, receiver address.Address, gasIpld cbornode.IpldStore, view FaultStateView, getter vmcontext.LookbackStateGetter) error {
+	if nv >= network.Version7 && blk.Height < -policy.ChainFinality {
```

Comment:
Here you are comparing reported fault epoch with negative Finality which will be always false.
It should probably be ‘currentEpoch - blk.Height < policy.ChainFinality’